vispy/testing/_runners.py

```diff
@@ -33,24 +33,20 @@ def _get_import_dir():
     return import_dir, dev
 
 
-_unit_script = """
-import pytest as tester
-try:
-    import faulthandler
-    faulthandler.enable()
-except Exception:
-    pass
-
-raise SystemExit(tester.main(%r))
-"""
-
-
-def _unit(mode, extra_arg_string, coverage=False):
+def _unit(mode, extra_arg_string='', coverage=False):
     """Run unit tests using a particular mode"""
+    if isinstance(extra_arg_string, str):
+        if len(extra_arg_string):
+            extra_args = extra_arg_string.split(' ')
+        else:
+            extra_args = ()
+    assert isinstance(extra_arg_string, (list, tuple))
+    assert all(isinstance(e, str) for e in extra_arg_string)
```

Comment:
Shouldn't these two asserts be on `extra_args`? Is that why @kmuehlbauer's tests were failing?

Reply:
Yes feel free to push

Reply:
@djhoese :+1:  Good catch, changing `extra_arg_string` to `extra_args` fixes this for me (tested with `python make test unit`)

Reply:
Pushed. Feel free to try cases 1-4 from #1571. I only tested (3) (and now a bit `pip install -e .` as well, which isn't in the list)